swan-lake/resources/featured-scenarios/call-llms-with-natural-expressions.md

```diff
@@ -0,0 +1,287 @@
+    resource function get attractions(string country, string interest, int count = 5) 
+            returns Attraction[]|http:InternalServerError {
+
+        Attraction[]|error attractions = natural {
+            Tell me the top ${count} places to visit in ${country} which are 
+            good for a tourist who has an interest in ${interest} to visit.  
+            Include a highlight one-liner about that place.
+        };
+
+        if attractions is Attraction[] {
+            return attractions;
+        }
+        return {body: "Failed to fetch attractions"};
+    }
+}
+```
+
+## The `service.bal` file complete code 
+
+The complete code in the `service.bal` will be as follows.
+
+```ballerina
+import ballerina/http;
+
+# Represents a tourist attraction.
+type Attraction record {|
+    # The name of the attraction
+    string name;
+    # The city where the attraction is located
+    string city;
+    # A notable feature or highlight of the attraction
+    string highlight;
+|};
+
+service on new http:Listener(8080) {
+    resource function get attractions(string country, string interest, int count = 5) 
+            returns Attraction[]|http:InternalServerError {
+        Attraction[]|error attractions = natural {
+            Tell me the top ${count} places to visit in ${country} which are 
+            good for a tourist who has an interest in ${interest} to visit.  
+            Include a highlight one-liner about that place.
+        };
+
+        if attractions is Attraction[] {
+            return attractions;
+        }
+        return {body: "Failed to fetch attractions"};
+    }    
+}
+```
+
+## Run the service 
+
+Execute the command below to run the service.
+
+```
+$ bal run --experimental
+```
+
+You view the output below.
+
+```
+Compiling source
+        ballerina_tutorials/attraction_finder_service:0.1.0
+
+Running executable
+```
+
+## Configure the LLM
+
+### Configure the default LLM
+
+Provide configuration for the LLM via configurable variables. This configuration will be used in an LLM client across all natural expressions, unless a model is explicitly specified in the natural expression.
+
+You can use your keys and configuration for OpenAI or Azure OpenAI by configuring the `defaultModelConfig` variable in the `ballerina/np` module.
+
+E.g., add the following for Azure OpenAI in the Config.toml file
+
+```toml
+[ballerina.np.defaultModelConfig]
+serviceUrl = "<SERVICE_URL>"
+deploymentId = "<DEPLOYMENT_ID>"
+apiVersion = "<API_VERSION>"
+connectionConfig.auth.apiKey = "<YOUR_API_KEY>"
+```
+
+Alternatively, you can use the default model made available via WSO2 Copilot. Log in to WSO2 Copilot, open up the VS Code command palette (`Ctrl + Shift + P` or `command + shift + P`), and run `Configure Default Model for Natural Functions`. This will add configuration for the default model into the Config.toml file. Please note that this will require VS Code being open in the relevant directory.
```

Comment:
Dont we need to mention this step before the `Run the service` step?

Reply:
Yeah, this was a mistake, meant to add it before run.